### ISSUE TITLE
Overwriting font-size styles in multiple CSS classes fails on Android

### ISSUE THREAD

**Issue by Buuhuu** [NativeScript/NativeScript#4234]
Assuming we have the following style defined:

```
.classA {
 font: 31 'icon-font';
}

Label.classB {
 font: 31 Arial
}
```

and the following template

```
<Label class="classA classB" text="Text"><Label>
```

This is giving me the following exception on *Android* using tns-core-modules *3.0.0*. 

```
JS: ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'nativeSize' of undefined
JS: TypeError: Cannot read property 'nativeSize' of undefined
JS:     at Label.TextBase.(anonymous function) (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/text-base/text-base.js:137:91)
JS:     at Style.css:font-size (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/properties/properties.js:649:46)
JS:     at Object.resetCSSProperties (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/properties/properties.js:885:40)
JS:     at CssState.apply (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/styling/style-scope.js:91:22)
JS:     at file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:306:29
JS:     at Label.ViewBase._batchUpdate (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:216:20)
JS:     at Label.ViewBase.applyCssState (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:300:14)
JS:     at Label.ViewBase._setCssState (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:293:14)
JS:     at StyleScope.applySelectors (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/styling/style-scope.js:228:14)
JS:     at Label.ViewBase._applyStyleFromScope (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:233:19)
JS:     at resetStyles (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:699:10)
JS:     at file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:701:9
JS:     at GridLayout.LayoutBaseCommon.eachChildView (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/layouts/layout-base-common.js:125:26)
JS:     at GridLayout.ViewCommon.eachChild (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view/view-common.js:690:14)
JS:     at resetStyles (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:700:10)
JS:     at valueChanged (file:///data/data/ch.myapp.nc/files/app/tns_modules/tns-core-modules/ui/core/view-base/view-base.js:694:9)
```

From the stacktrace we can see that when applying the Style first `resetCSSProperties` is called setting the `defaultValue` to the `View`, which in that case is `undefined` and has no property `nativeSize`

See https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/ui/styling/style-properties.ts#L932

I have no clue why that is working properly on iOS.

Note: Worked with 2.5.x

**Comment by NickIliev** [comment]
@Buuhuu I have tried the described case with [this test app](https://github.com/NickIliev/NativeScript-Issues-2017/tree/master/NativeScript/issue_4234) and on my side, the style is applied successfully.

If you are still experiencing this please provide basic application, that can reproduce the issue so we can test it with your code.

**Comment by Buuhuu** [comment]
You are right. As you can see from the Stacktrace the `Label` is in `GridLayout` and the call is triggered by `valueChanged` so its in my case Angular's change detection changing the `className`

See the sample here: 

https://github.com/Buuhuu/nativescript-issue-4234

**Comment by Buuhuu** [comment]
@NickIliev why did you close that without resolving the issue? For us that caused a bad experience migrating while upgrading to 3.0. Others might come across the same at some point so it should be resolved.

**Comment by NickIliev** [comment]
Hey @Buuhuu as far as I understood there is no issue - based on your example application the behavior is exactly the same on both iOS and Android (the default style is loading and then after the setTimeout the additional style is triggered).

Of course, I will reopen this one but please clarify for us what is the unexpected behavior for Android.

**Comment by Buuhuu** [comment]
Ok I cannot reproduce it either, which is weird as the same still happens when I remove my patch form the actual App we are working on. Though its bit too complex to easily "build an sample of". 

What I found out while debugging is that [properties.ts#L115](https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/ui/core/properties/properties.ts#L115) is called before [properties.ts#L126(https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/ui/core/properties/properties.ts#L126) for a particular combination of css classes. Any idea why that could be the case?

**Comment by vakrilov** [comment]
Hi @Buuhuu 
The code in the [properties.ts](https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/ui/core/properties/properties.ts) is executed for most of the properties of the NativeScript views. It is hard to tell what is going on without a stable reproduce of the error.

Are you still having problems this issue in your app with the `3.0.1` release? There was a fix(#4159) that might be related and might solve this issue too. 

I tried, but I was unable to reproduce a crash or see any exceptions in the logs with your sample app (both with `3.0.0` and `3.0.1`). If you are still having a problem with your original app - please try to isolate it (as much as possible) and we can then work on a resolution.

**Comment by NickIliev** [comment]
@Buuhuu, as discussed above, the issue is not reproducible so I am closing this one due to inactivity. Please do let us know if you came across this one again.
